designs/component-upgrades.md

```diff
@@ -0,0 +1,149 @@
+# EKS-A components upgrade
+
+
+## Introduction
+
+**Problem:** customers are only able to upgrade the cluster components that are part of the kubernetes distribution. Any other core component of EKS-A remains untouched from the cluster creation.
+
+This stops customers from enjoying the latest features and bug fixes from eks-a and any of its core components as well as from being able to get security patches.
+
+### Tenets
+
+* **Simple:** simple to use, simple to maintain, simple to build
+
+### Goals and objectives
+
+As an eks-a customer:
+
+* I want to able to enjoy the latest eks-a features and bug fixes
+* I want to keep my cluster up to date with security patches
+* I don’t want to need to stay informed about latest versions, security patches or compatibility between components
+
+### Statement of scope
+
+**In scope**
+
+* Upgrade core eks-a components to latest versions available for an specific cli version
+    * Core CAPI
+    * CAPI providers
+    * Etcdadm CAPI provider
+    * CNIs (cilium)
+    * EKS-A controller and CRDs
+    * Flux
+
+**Not in scope**
+
+* Install/upgrade add-ons
+* Upgrade cli
+* Upgrades to customer defined component versions
+
+**Future scope**
+
+*  Version changes report
+* “Notification” for new available versions
+* Automatic/scheduled upgrades
+
+## Overview of solution
+
+**Upgrade all eks-a components to the versions declared in the latest bundles manifest when running the `upgrade cluster` command.**
+
+## Solution details
+
+Since we already have:
+
+* An interface that customers use to upgrade clusters (`upgrade cluster` command)
+* An declarative list of all eks-a components at their latest available versions that have been tested together (bundles manifest)
+
+It seems like the simplest solution is to take advantage of both.
+
+This solution fits very well with both, the goal of a customer experience as simple as possible and the concept of the versions bundle (a group of versioned components that work well together):
+
+* We don’t introduce more concepts/commands. Customers just have to use the same command they are already used to. The syntax stays the same and there is no need to define what needs to be upgraded and to what version, the cli takes care of this transparently.
+* Components bundled together are always upgraded together. This reduces our testing matrix considerably, since there is no room for version miss-alignments.
```

Comment:
I would call out that another benefit is that customers technically never have to worry about the individual version and dependencies, which also a benefit of this project.